pkg/ccl/backupccl/backup_test.go

```diff
@@ -10213,3 +10214,136 @@ func TestBackupDBWithViewOnAdjacentDBRange(t *testing.T) {
+	for i := 2; i < 1002; i++ {
+		db.Exec(t, `UPDATE foo SET id = $1`, i)
+	}
+	db.Exec(t, `SET CLUSTER SETTING kv.bulk_sst.max_allowed_overage = '19b'`)
+	db.Exec(t, `SET CLUSTER SETTING kv.bulk_sst.target_size = '1b'`)
+
+	// Each response is roughly 940bytes large. 845100b ensures that we have ~900
+	// revisions in our file buffer that we then sort before writing to the
+	// underlying SSTWriter. This configuration was sufficient to result in an
+	// unstable sort when the test was run under stress.
+	db.Exec(t, `SET CLUSTER SETTING bulkio.backup.merge_file_buffer_size = '845100b'`)
+	db.Exec(t, `BACKUP TABLE foo INTO 'nodelocal://1/foo' WITH revision_history`)
+}
+
+// TestSSTQueueSort is a unit test for the sort algorithm of the `sstQueue`
+// type.
+func TestSSTQueueSort(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	for _, tc := range []struct {
+		name  string
+		elems sstQueue
+	}{
+		{
+			"simple",
+			sstQueue{
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+					},
+				},
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("c"), EndKey: roachpb.Key("d")},
+					},
+				},
+			},
+		},
+		{
+			"start-key-equal-end-key-unequal",
+			sstQueue{
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("a")},
+					},
+				},
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+					},
+				},
+			},
+		},
+		{
+			"start-key-equal-end-key-equal-walltime",
+			sstQueue{
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("a")},
+					},
+					endKeyTS: hlc.Timestamp{WallTime: 2, Logical: 1},
+				},
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("a")},
+					},
+					endKeyTS: hlc.Timestamp{WallTime: 1, Logical: 7},
+				},
+			},
+		},
+		{
+			"start-key-equal-end-key-equal-logical",
+			sstQueue{
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("a")},
+					},
+					endKeyTS: hlc.Timestamp{WallTime: 1, Logical: 7},
+				},
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("a")},
+					},
+					endKeyTS: hlc.Timestamp{WallTime: 1, Logical: 1},
+				},
+			},
+		},
+		{
+			"start-key-equal-end-key-equal-empty-endkeyTS",
+			sstQueue{
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("a")},
+					},
+					endKeyTS: hlc.Timestamp{WallTime: 1, Logical: 7},
+				},
+				{
+					f: BackupManifest_File{
+						Span: roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("a")},
+					},
+				},
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			var queue sstQueue
+			queue = append(queue, tc.elems...)
```

Comment:
If you wanted, I suppose we could randomize the order of these

Reply:
good idea, done